test/distributed/org/apache/cassandra/distributed/test/ReadRepairQueryTypesTest.java

```diff
@@ -0,0 +1,1200 @@
+                            rows(row(0, null, list(3, 4))))
+                .tearDown();
+    }
+
+    /**
+     * Test unrestricted queries with frozen maps.
+     */
+    @Test
+    public void testFrozenMap()
+    {
+        tester().withQueryRestriction("")
+                .createTable("CREATE TABLE %s (k int PRIMARY KEY, a frozen<map<int,int>>, b frozen<map<int,int>>)")
+                .populateNode("INSERT INTO %s (k, a, b) VALUES (0, {1:10, 2:20}, {3:30, 4:40})")
+                .queryColumns("a[2]",
+                              rows(row(20)),
+                              rows(row(0, map(1, 10, 2, 20), map(3, 30, 4, 40))),
+                              rows(row(0, map(1, 10, 2, 20), null)))
+                .deleteColumn("DELETE a FROM %s WHERE k=0",
+                              "b[4]",
+                              rows(row(40)),
+                              rows(row(0, null, map(3, 30, 4, 40))),
+                              rows(row(0, map(1, 10, 2, 20), map(3, 30, 4, 40))))
+                .deleteRows("DELETE FROM %s WHERE k=0",
+                            rows(),
+                            rows(row(0, null, map(3, 30, 4, 40))))
+                .tearDown();
+    }
+
+    /**
+     * Test unrestricted queries with frozen user-defined types.
+     */
+    @Test
+    public void testFrozenUDT()
+    {
+        tester().withQueryRestriction("")
+                .createTable("CREATE TABLE %s (k int PRIMARY KEY, a frozen<udt>, b frozen<udt>)")
+                .populateNode("INSERT INTO %s (k, a, b) VALUES (0, {x:1, y:2}, {x:3, y:4})")
+                .queryColumns("a.x",
+                              rows(row(1)),
+                              rows(row(0, udt("x", 1, "y", 2), udt("x", 3, "y", 4))),
+                              rows(row(0, udt("x", 1, "y", 2), null)))
+                .deleteColumn("DELETE a FROM %s WHERE k=0",
+                              "b.y",
+                              rows(row(4)),
+                              rows(row(0, null, udt("x", 3, "y", 4))),
+                              rows(row(0, udt("x", 1, "y", 2), udt("x", 3, "y", 4))))
+                .deleteRows("DELETE FROM %s WHERE k=0",
+                            rows(),
+                            rows(row(0, null, udt("x", 3, "y", 4))))
+                .tearDown();
+    }
+
+    /**
+     * Test unrestricted queries with non-frozen sets.
+     */
+    @Test
+    public void testNonFrozenSet()
+    {
+        tester().withQueryRestriction("")
+                .createTable("CREATE TABLE %s (k int PRIMARY KEY, a set<int>, b set<int>, c int)")
+                .populateNode("INSERT INTO %s (k, a, b, c) VALUES (0, {1, 2}, {3, 4}, 10)")
+                .queryColumns("a[1]",
+                              rows(row(1)),
+                              rows(row(0, set(1, 2), set(3, 4), 10)),
+                              rows(row(0, set(1), set(3, 4), null))) // unselected 'b' is repaired, see CASSANDRA-10657
```

Comment:
@maedhroz here the complex column `b` is read-repaired even though it's not selected by the query. This happens in [`ColumnFilter#fetchedCellIsQueried`](https://github.com/apache/cassandra/blob/474de80de5179e51fcc5b125c94ef68111c63c90/src/java/org/apache/cassandra/db/filter/ColumnFilter.java#L240-L242) since at least CASSANDRA-10657. The same happens for other complex column types below. Not sure this repair is necessary, but at least it doesn't seem to break anything, so we could take a closer look at it on a followup ticket, as a performance improvement.